src/Features/CSharp/Portable/ExtractInterface/CSharpExtractInterfaceService.cs

```diff
@@ -33,22 +34,22 @@ public CSharpExtractInterfaceService()
 
         protected override async Task<SyntaxNode> GetTypeDeclarationAsync(Document document, int position, TypeDiscoveryRule typeDiscoveryRule, CancellationToken cancellationToken)
         {
-            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-            var root = await tree.GetRootAsync(cancellationToken).ConfigureAwait(false);
-            var token = root.FindToken(position != tree.Length ? position : Math.Max(0, position - 1));
-            var typeDeclaration = token.GetAncestor<TypeDeclarationSyntax>();
+            var span = new TextSpan(position, 0);
+            var relevantNode = await document.TryGetRelevantNodeAsync<SyntaxNode>(span, cancellationToken).ConfigureAwait(false);
```

Comment:
Anyone know the perf cost of this call? Is it expensive? I could doTryGetRelevantNodeAsync<TypeDeclarationSyntax>. I wasn't sure if it is expensive to then call for the relevant SyntaxNode and parent in the case of TypeDiscoverRule.TypeDeclaration.